lib/view_model/active_record/update_operation.rb

```diff
@@ -426,7 +441,7 @@ def build_updates_for_collection_association(association_data, association_updat
               end
 
             when FunctionalUpdate::Remove
-              removed_refs = Set.new(fupdate.values.map(&:viewmodel_reference))
+              removed_refs = fupdate.values.to_set
```

Comment:
It used to be acceptable for `Remove` types to have `UpdateData`s, which was misleading at best, since they'd never actually be part of the update op tree, and were constrained not to have any actual updates. In shared land we're much more careful about orphaned updates, so this was necessary, and I think clearer.